paasta_tools/metrics/metastatus_lib.py

```diff
@@ -377,18 +389,44 @@ def filter_tasks_for_slaves(slaves, tasks):
     return [task for task in tasks if task['slave_id'] in slave_ids]
 
 
-def get_resource_utilization_by_grouping(grouping_func, mesos_state):
+def filter_slaves(slaves, filters):
```

Comment:
can you make this more generic? make filters be a list of predicate fns that match on the slave object and do something like:

```
return [s for s in slaves if all[f(s) for f in filters]]
```